py-polars/polars/expr/expr.py

```diff
@@ -7424,11 +7424,11 @@ def kurtosis(self, *, fisher: bool = True, bias: bool = True) -> Self:
         """
         return self._from_pyexpr(self._pyexpr.kurtosis(fisher, bias))
 
-    def clip(self, lower_bound: int | float, upper_bound: int | float) -> Self:
+    def clip(self, lower_bound: IntoExpr, upper_bound: IntoExpr) -> Self:
```

Comment:
I think we need a type alias for "something representing an expression column", i.e.:

```python
IntoExprColumn: TypeAlias = Union["Expr", "Series", str]
```

Together with `PythonLiteral` and `None`  this will form the `IntoExpr` type.

And for some functions, you can define a subset of `PythonLiteral` if they do not take all possible literals.

Reply:
See #11296

Will merge it when CI passes, then you can update this PR if you want.

Reply:
Merged!

Reply:
I guess you could add `NumericLiteral` for (int, float, Decimal) and `TemporalLiteral` for (time, date, datetime, timedelta). Then you could use those?